### DISCUSSION TITLE
why service_traffic not found in Graphql

### DISCUSSION THREAD

**Discussion by tunbb** [apache/skywalking#12070] [Q&A]
I RUN ES 7.11.2 AND SKYWALKING 9.5.5
SET A PROMETHEUS DATASOURCE: skywalking
but I cannot get service_traffic metric,why? then i cannot get all svc

**Comment by wu-sheng** [comment]
It is not a metric, you should use metadata v2 query protocol, https://github.com/apache/skywalking-query-protocol/blob/master/metadata-v2.graphqls

**Reply by tunbb** [reply]
oh~ I found the example "/api/v1/series?match[]=service_traffic{layer='GENERAL'}&start=1677479336&end=1677479636" in skywalking graphql api doc, https://skywalking.apache.org/docs/main/v9.5.0/en/api/promql-service/#sampled-records 
it misleaded me .
BTW, why dashboard use it as metrics here? https://skywalking.apache.org/docs/main/next/en/setup/backend/ui-grafana/
![image](https://github.com/apache/skywalking/assets/119464933/54f2b9fd-d51b-49c8-8831-7960e873771c)
thanks for reply!

**Reply by wu-sheng** [reply]
The deep why is about how to make PromQL concepts work with SkyWalking concepts. The root cause is, there is no service concept in PromQL or Grafana, only metrics. But in SkyWalking, they are very different. 
The Grafana dashboard you are referring is actually not using GraphQL, it is using PromQL APIs, which docs are provided below.

You seems to mix these two things, they are very different.

**Reply by wu-sheng** [reply]
PromQL API implementation is an API adaptor on the top of GraphQL.

**Reply by tunbb** [reply]
Yes, I mix them. But I still don't understand why dashboard(General Service ) can use "service_traffic{layer="$layer"} to get all services but it shows nothing. I follow your link below and add an datasource, when I search service_traffic it shows "no metrics found". 
I just want the service_traffic can work on the General Service dashboard, what should I do?
![image](https://github.com/apache/skywalking/assets/119464933/1e81e7b6-2c97-4609-979b-4b8f9bc38102)

**Reply by wu-sheng** [reply]
I can't tell what is your side issue, but you could check https://skywalking.apache.org/#demo, there is Grafana deployment which is working and including all configurations.

**Reply by tunbb** [reply]
Generally， There is no special configuration which exposes service_traffic right?
I will try the demo,Thank you!

**Reply by wu-sheng** [reply]
There isn't special thing like that

**Comment by wu-sheng** [comment]
> SET A PROMETHEUS DATASOURCE: skywalking

PromQL doc is here, https://skywalking.apache.org/docs/main/latest/en/api/promql-service/
And Grafana setup is here, https://skywalking.apache.org/docs/main/latest/en/setup/backend/ui-grafana/
